pennylane/circuit_drawer.py

```diff
@@ -0,0 +1,916 @@
+
+        Args:
+            layer (list): Layer that will be appended
+        """
+        self.raw_grid_transpose.append(layer)
+        self.raw_grid = _transpose(self.raw_grid_transpose)
+
+        return self
+
+    def replace_layer(self, idx, layer):
+        """Replace a layer in the Grid at the specified index.
+
+        Args:
+            idx (int): Index of the layer to be replaced
+            layer (list): Layer that replaces the old layer
+        """
+        self.raw_grid_transpose[idx] = layer
+        self.raw_grid = _transpose(self.raw_grid_transpose)
+
+        return self
+
+    def insert_wire(self, idx, wire):
+        """Insert a wire into the Grid at the specified index.
+
+        Args:
+            idx (int): Index at which to insert the new wire
+            wire (list): Wire that will be inserted
+        """
+        self.raw_grid.insert(idx, wire)
+        self.raw_grid_transpose = _transpose(self.raw_grid)
+
+        return self
+
+    def append_wire(self, wire):
+        """Append a wire to the Grid.
+
+        Args:
+            wire (list): Wire that will be appended
+        """
+        self.raw_grid.append(wire)
+        self.raw_grid_transpose = _transpose(self.raw_grid)
+
+        return self
+
+    @property
+    def num_layers(self):
+        """Number of layers in the Grid.
+
+        Returns:
+            int: Number of layers in the Grid
+        """
+        return len(self.raw_grid_transpose)
+
+    def layer(self, idx):
+        """Return the layer at the specified index.
+
+        Args:
+            idx (int): Index of the layer to be retrieved
+
+        Returns:
+            list: The layer at the specified index
+        """
+        return self.raw_grid_transpose[idx]
+
+    @property
+    def num_wires(self):
+        """Number of wires in the Grid.
+
+        Returns:
+            int: Number of wires in the Grid
+        """
+        return len(self.raw_grid)
+
+    def wire(self, idx):
+        """Return the wire at the specified index.
+
+        Args:
+            idx (int): Index of the wire to be retrieved
+
+        Returns:
+            list: The wire at the specified index
+        """
+        return self.raw_grid[idx]
+
+    def copy(self):
+        """Create a copy of the Grid.
+
+        Returns:
+            Grid: A copy of the Grid
+        """
+        return Grid(self.raw_grid.copy())
+
+    def append_grid_by_layers(self, other_grid):
+        """Append the layers of another Grid to this Grid.
+
+        Args:
+            other_grid (pennylane.circuit_drawer.Grid): Grid whos layers will be appended
+        """
+        for i in range(other_grid.num_layers):
+            self.append_layer(other_grid.layer(i))
+
+        return self
+
+    def __str__(self):
+        """String representation"""
+        ret = ""
+        for wire in self.raw_grid:
+            ret += str(wire)
+            ret += "\n"
+
+        return ret
+
+
+class CharSet(abc.ABC):
```

Comment:
I just don't want people to instantiate it, and I hoped that inheriting from `ABC` would realize that.

Reply:
I think this will still be instantiable. `abc.ABC` only blocks abstract classes from being instantiated if decorated abstract methods exist that aren't being overwritten.

Consider the following:

```python
import abc

class Test(abc.ABC): 
    pass

class Test2(abc.ABC):

    @abc.abstractmethod
    def must_overwrite(self):
        pass
```

The first class will be instantiable, whereas the second class won't be:
```python
>>> t = Test()
>>> t = Test2()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-6-0e60783dfdf4> in <module>
----> 1 t = Test2()

TypeError: Can't instantiate abstract class Test2 with abstract methods must_overwrite
```

Without abstract methods/properties, this is really more of a standard base class as opposed to an abstract base class (in Python terminology)